jax/_src/lax/linalg.py

```diff
@@ -1188,10 +1188,30 @@ def svd_jvp_rule(primals, tangents, full_matrices, compute_uv):
     dV = dV + jnp.matmul(jnp.eye(n, dtype=A.dtype) - jnp.matmul(V, Vt), jnp.matmul(_H(dA), U)) / s_dim
   return (s, U, Vt), (ds, dU, _T(dV))
 
-def _svd_cpu_gpu_translation_rule(gesvd_impl, c, operand, full_matrices, compute_uv):
+def _empty_svd(a, *, full_matrices, compute_uv):
+  batch_shape = a.shape[:-2]
+  m, n = a.shape[-2:]
+  s = jnp.empty(batch_shape + (0,), dtype=lax_internal._complex_basetype(a.dtype))
+  if not compute_uv:
+    return (s,)
+  if full_matrices:
+    size = max(m, n)
+    u = jnp.broadcast_to(jnp.eye(size, dtype=a.dtype), batch_shape + (size, size))
+  else:
+    u = jnp.empty(batch_shape + (m, n), dtype=a.dtype)
+  v = jnp.empty(batch_shape + (0, 0), dtype=a.dtype)
+  if m < n:
+    u, v = v, u
+  return s, u, v
 
+def _svd_cpu_gpu_translation_rule(gesvd_impl, empty_ok, c, operand, full_matrices, compute_uv):
   shape = c.get_shape(operand)
   batch_dims = shape.dimensions()[:-2]
+
+  if not empty_ok and prod(shape.dimensions()[-2:]) == 0:
```

Comment:
I'd actually be tempted to unconditionally use this code path for empty arrays, even on GPU. It would simplify the code a little.

Reply:
Thanks - that makes sense. I'll make that change.